### PR TITLE
feat: Add monorepo support

### PR DESCRIPTION
**Pull request by Abraxas-365**
Several modifications have been made across files to enable support for monorepos. This includes changes to the app service and initialization command, and the introduction of prompts to gather user input for monorepo setup.

### CODE REVIEW

File: cmd/initialize/initlizeMonorepo.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+			return handleErr
+		}
+		if existingApp == nil {
+			return fmt.Errorf("operation cancelled for app: %s", appName)
+		}
+
+		newApp = *existingApp
+	}
+
+	// Rest of the function remains the same...
```
Comment (kylekampy) [inline cmd/initialize/initlizeMonorepo.go:142]: This comment seems out of place.

File: cmd/initialize/initlizeMonorepo.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+		}
+		version = *envStruct.Version
+	}
+
+	db, err := database.Restore()
+	if err != nil {
+		return err
+	}
+
+	newEnvDcrypted, err := envStruct.DecryptData(m.GetSecretKey())
```
Comment (kylekampy) [inline cmd/initialize/initlizeMonorepo.go:238]: Typo on `Decrypted`

File: cmd/initialize/initlizeMonorepo.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+	}
+
+	if err := db.UpsertSecret(secretKey, newEnvDcrypted, version); err != nil {
+		return fmt.Errorf("failed to save local environment: %w", err)
+	}
+
+	return nil
+}
+
+func CreateGitignoredFileMonorepo(cmd *cobra.Command, fullPath, fileName string) error {
```
Comment (kylekampy) [inline cmd/initialize/initlizeMonorepo.go:256]: This code looks nearyl identical to the CreateGitignoredFile I commented on before. I think it'd make sense to consolidate it in gitutils.

**Review by kylekampy — COMMENTED** [review]
I've got a handful of comments I'd love to see fixed. But overall I think the code looks good. As long as the UX fits with what you've discussed with Matthew, I'm fine with approving this.

File: cmd/initapp/initapp.go [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+		return err
+	}
+
+	secretKey := database.SecretKey{
+		ProjectId: *m.ProjectId,
+		AppId:     *m.AppId,
+		EnvName:   envName,
+	}
+
+	if err := db.UpsertSecret(secretKey, newEnvDcrypted, version); err != nil {
```
[the remark this answers is not in the document]
Comment (Abraxas-365) [reply cmd/initapp/initapp.go:270]: This is in the global .hx, we had a conversation about this like 2 moths ago, when Matthew told us, maybe in the future we should  use a db

File: cmd/initialize/initlizeMonorepo.go (hunk 3 above)
Comment (Abraxas-365) [reply cmd/initialize/initlizeMonorepo.go:255]: I would like to have the monorepo one and the normal one. In the future, they could do more different things.

**Pull request merged**